index.js

```diff
@@ -243,9 +271,49 @@ Shopify.prototype.graphql = function graphql(data, variables) {
     parseJson: this.options.parseJson,
     timeout: this.options.timeout,
     responseType: 'json',
-    retry: 0,
     method: 'POST',
-    body: json ? this.options.stringifyJson({ query: data, variables }) : data
+    body: json ? this.options.stringifyJson({ query: data, variables }) : data,
+    retry:
+      this.options.maxRetries > 0
+        ? {
+            limit: this.options.maxRetries,
+            // Don't clamp Shopify retry-after header values too low.
+            maxRetryAfter: Infinity,
+            calculateDelay: (retryObject) => {
+              attemptNumber += 1;
+              return (
+                graphQLRetryMS(
+                  attemptNumber,
+                  this.options.maxRetries,
+                  retryObject.error
+                ) || retryObject.computedValue
+              );
+            }
+          }
+        : 0,
+    hooks: {
+      afterResponse:
+        this.options.maxRetries > 0
+          ? [
+              (response) => {
+                if (response.body && response.body.errors) {
+                  const retryDelay = graphQLRetryMS(
+                    attemptNumber,
+                    this.options.maxRetries,
+                    { response }
+                  );
+
+                  if (retryDelay) {
+                    // Make got consider this response errored and retry
+                    throw new Error('GraphQL cost limit exceeded, retrying');
```

Comment:
I would create an error like it is done below starting on line 325, so that error is used if the maximum number of attempts is reached.